extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -206,6 +217,31 @@ protected TaskStatus doTask(Task task, boolean run)
     }
   }
 
+  @VisibleForTesting
+  protected void listenOnTaskState(KubernetesPeonLifecycle.State state, String taskId)
+  {
+    switch (state) {
+      case RUNNING:
+        KubernetesWorkItem workItem;
+        synchronized (tasks) {
+          workItem = tasks.get(taskId);
+          if (workItem == null) {
+            log.error("Task [%s] disappeared", taskId);
+            return;
+          }
+        }
+        ServiceMetricEvent.Builder metricBuilder = new ServiceMetricEvent.Builder();
+        IndexTaskUtils.setTaskDimensions(metricBuilder, workItem.getTask());
+        emitter.emit(metricBuilder.build(
+                         "task/pending/time",
+                         new Duration(workItem.getCreatedTime(), DateTimes.nowUtc()).getMillis()
+                     )
+        );
+      default:
+        // ignore other state transition now
```

Comment:
Nit: Seems weird to not have any statement here 😅 , maybe just add a `return;`

Reply:
Added. But IntelliJ will flag it as unnecessary, let's see if github checks are good.